### PR TITLE
fix(#1359): prevent focus-trap from allowing element with a tabindex=…

### PR DESCRIPTION
**Pull request by chrisolsen**
…-1 from attaining focus

### CODE REVIEW

**Comment by chrisolsen** [comment]
Can be tested with the following. Where the last button should not be focusable (previous was when shift-tabbing)
```
    <goa-popover>
      <goa-button slot="target">Click me</goa-button>
      This is the popover content
      <button tabindex="0">Close</button>
      <button tabindex="0">Close</button>
      <button tabindex="0">Close</button>
      <button tabindex="-1">Close</button>
    </goa-popover>
```

File: libs/web-components/src/components/focus-trap/FocusTrap.spec.ts [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+  cleanup,
+  fireEvent,
+  createEvent,
+  waitFor,
+} from "@testing-library/svelte";
+import FocusTrapTestComponent from "./FocusTrapTestComponent.svelte";
 
 afterEach(cleanup);
 
 // This test is blocked due to JEST handling slot elements differently than web browser
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/focus-trap/FocusTrap.spec.ts:13]: You've modified this test, and removed the skip. Is this comment still applicable?

Comment (chrisolsen) [reply libs/web-components/src/components/focus-trap/FocusTrap.spec.ts:13]: The test was modified in an attempt to get it working and the skip was just moved, but it's still there.

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**